### DISCUSSION TITLE
find path statement without limit not available

### DISCUSSION THREAD

**Discussion by jxnu-liguobin** [vesoft-inc/nebula#5710] [Q&A]
I waited on the nebula studio for a while, but there were no errors or data displays.
I saw an error on the server:

<img width="988" alt="image" src="https://github.com/vesoft-inc/nebula/assets/29496873/2d282503-a172-40ea-81e4-5999529a7f81">

If I reduce the upto or add the limit, it will work normally. 
My machine has 80GB of memory. About 600000 points, two million edges.

My ngql is `FIND NOLOOP PATH WITH PROP FROM "926468220071731200" TO "919758682291462144" OVER * UPTO 100 STEPS YIELD path as p `

Is there a way for me to correctly query the data?

**Comment by wey-gu** [comment]
Should be killed by Memory Tracker as the first line of the log. 

For now in NebulaGraph 3.x:

1. Path construction doesn't compress RAM in a way that doesn't duplicate usage from the same node/edge for a whole-graph-like traversal
2. For the nature of most of the graph, 10 steps would end up covering the whole graph's data
3. The UDF error line should only be shown in the process booting of graphd, I suspect the graphd got restarted (not quite sure)

For graph analytics like 100 steps of the path, it's recommended to be done in NebulaGraph algorithm/ graphx/nebulagraph analytics. :-D

**Comment by jxnu-liguobin** [comment]
> 3\. The UDF error line should only be shown in the process booting of graphd, I suspect the graphd got restarted (not quite sure)

Thank you.
Yes, the graphd service continues to work well. 

> For graph analytics like 100 steps of the path, it's recommended to be done in NebulaGraph algorithm/ graphx/nebulagraph analytics. :-D

I will give it a try.

**Reply by wey-gu** [reply]
As I know NebulaGraph Algorithm out of the box only supports the shortest path, for all path, we may need to add our own algos,but it won't be a big trouble for you(the scala hacker) as NebulaGraph Algorithm is GraphX/Spark based and it speaks............. SCALA

see https://github.com/vesoft-inc/nebula-algorithm/tree/master/nebula-algorithm/src/main/scala/com/vesoft/nebula/algorithm/lib

**Reply by jxnu-liguobin** [reply]
The only problem is that this requires a Spark,but we currently do not really need Spark
